zebrad/tests/acceptance.rs

```diff
@@ -53,12 +54,21 @@ fn generate_args() -> Result<()> {
     let output = child.wait_with_output()?;
     output.assert_failure()?;
 
+    // create config file path in the same location as binary
+    let binary_path = Path::new(env!("CARGO_BIN_EXE_zebrad"));
```

Comment:
Issue: Use the temporary directory for each test, not the shared build directory.

Using the shared build directory can lead to file read/write conflicts.

Here's how you can access the temporary directory:
```suggestion
    let tempdir_path = _guard.path();
```

`_guard` is actually a `TempDir` instance:
https://docs.rs/tempdir/0.3.7/tempdir/struct.TempDir.html#method.path

Suggestion: rename `_guard` to `tempdir`.

Reply:
I actually tried something like this but i had a problem. `_guard` is not just an instance of `TempDir` but a `impl std::ops::Drop` type that i don't know how to cast to a tempdir or path.  Suggested code ` let tempdir_path = _guard.path();` will fail to build.

One option that i tried is to return a `TempDir` third argument of `test_cmd`, that will work but ugly.

Reply:
I think you're on the right path, we're going to have to tweek the return type to a concrete type rather than an existential so we can call fns on it. If you'd like you can try cleaning up the API by wrapping the tempdir's guard with new_type that provides a nicer interface

Something like:

```rust
struct ZebraTestDir {
    tempdir: TempDir,
}

impl ZebraTestDir {
    fn with_config(&self, cmd: &mut Command, config: Config) -> Result<()> {
        // write config to the directory and add the correct args to the `Command` to get it to use the config.
    }
}
```

Reply:
I added this at https://github.com/ZcashFoundation/zebra/pull/859/commits/3b8925405f84b153540ef0e36f3318bd567de703 however i am not totally convinced but pushed. Maybe with a re review of you guys we can make something decent. Thanks.

Reply:
Please hold your review until tomorrow. Need adjustments.